Kuzzle.Tests/API/Controllers/RealtimeControllerTest.cs

```diff
@@ -0,0 +1,163 @@
+using KuzzleSdk.API.Controllers;
+using Xunit;
+using Newtonsoft.Json.Linq;
+using System;
+using KuzzleSdk.API.Options;
+using KuzzleSdk.API;
+using Moq;
+
+namespace Kuzzle.Tests.API.Controllers
+{
+    public class RealtimeControllerTest
+    {
+        private readonly RealtimeController _realtimeController;
+        private readonly KuzzleApiMock _api;
+
+        public RealtimeControllerTest()
+        {
+            _api = new KuzzleApiMock();
+            _realtimeController = new RealtimeController(_api.MockedObject);
+        }
+
+        [Fact]
+        public async void CountAsyncTest()
+        {
+            string roomId = "A113";
+            Int32 count = 3;
+            _api.SetResult(new JObject { {
+                "result", new JObject { {
+                    "count", count } }
+                } }
+            );
+
+            Int32 res = await _realtimeController.CountAsync(roomId);
+
+            _api.Verify(new JObject {
+                { "controller", "realtime" },
+                { "action", "count" },
+                { "body", new JObject {{ "roomId", roomId }}}
+            });
+            Assert.Equal<Int32>(count, res);
+        }
+
+        [Fact]
+        public async void PublishAsyncTest()
+        {
+            string index = "an_index";
+            string collection = "to";
+            JObject message = new JObject { { "infinity", "and beyond" } };
+
+            await _realtimeController.PublishAsync(index, collection, message);
+
+            _api.Verify(new JObject{
+                {"index" , index },
+                {"collection" , collection },
+                {"controller" , "realtime" },
+                {"action" ,"publish" },
+                {"body" , message },
+            });
+        }
+
+        [Theory]
+        [
+            MemberData(nameof(MockGenerators.GenerateSubscribeOptions),
```

Comment:
(nitpicking) My view on this is that generators only used by one class should be made local to that class' test file. The global `MockGenerators` class should only contain generators that will be used by multiple classes, such as search options or filters (will be used in the future by the security controller).